File: python/array/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        numbers_map = dict()
        
        for i, n in enumerate(nums):
            diff = target - n
            
            if diff in numbers_map:
                return numbers_map[diff], i
            
            numbers_map[n] = i

File: go/array/1. Two Sum.go
func twoSum(nums []int, target int) []int {
    
    numbers_map := make(map[int]int)
    
    for i := 0; i < len(nums); i ++ {
        cur_number := nums[i]
        diff := target - cur_number
        _, found := numbers_map[diff]
        
        if found {
            return []int{numbers_map[diff], i}
        }
        numbers_map[cur_number] = i
    }
    return []int{}
}